rMATS_pipeline/rmatspipeline/rmatspipeline.pyx

```diff
@@ -2997,20 +3028,38 @@ cdef load_sg(str bams, const string& tmp, const int& rl,
     cdef:
         int num = 0, num_file = 0
         list vbams = bams.split(','), all_files
-        list flags
+        list prep_counts_by_bam
         vector[unordered_map[string,cmap[Tetrad,pair[int,int]]]] exons
         vector[unordered_map[string,cmap[vector[pair[long,long]],int]]] juncs
 
     num = len(vbams)
-    flags = [0 for i in range(num)]
+    prep_counts_by_bam = [0 for i in range(num)]
 
     all_files = [join(root, name) for root, dirs, files in walk(tmp)
                  for name in files if name.endswith('.rmats')]
     for name in all_files:
-        _load_job(name, vbams, flags, rl, novel_juncs, exons, juncs, sg_mode)
-    flags = [ele != 1 for ele in flags]
-    if any(flags):
-        print 'WARNING: there are redundant temporary files.'
+        _load_job(name, vbams, prep_counts_by_bam, rl, novel_juncs, exons, juncs, sg_mode)
+
+    any_error = False
+    for i, prep_count in enumerate(prep_counts_by_bam):
+        if prep_count == 0:
+            bam_name = vbams[i]
+            bam_count = vbams.count(bam_name)
+            if bam_count != 1:
+                sys.stderr.write('{} given {} times in input\n'.format(
+                    bam_name, bam_count))
+            else:
+                sys.stderr.write('{} not found in .rmats files\n'.format(
+                    vbams[i]))
```

Comment:
sys.stderr.write('{} not found in .rmats files\n'.format(vbams[i])) should be reported even when bam_count != 1. When bam_count != 1, two error messages should be both reported.

Reply:
Good catch! The "not found in .rmats" error would not be reported when it should have in some cases

I've pushed a new implementation that I think will report all the errors correctly. The previous implementation was checking the `bam_count` when `prep_count` was zero because `prep_counts_by_bam` is updated using the first index of the `bam_name` in `vbams`. That would cause all but the first entry in `prep_counts_by_bam` for a duplicate input bam to be zero even if that bam was in one of the .rmats files. The old implementation was attempting to avoid incorrectly reporting that the bam wasn't in any .rmats file when actually it was in a .rmats file but recorded at a different index in `prep_counts_by_bam`. The new implementation aggregates the information from `prep_counts_by_bam` using `bam_name` in one loop and then reports errors in separate loops